lib/common/options.c

```diff
@@ -317,9 +326,18 @@ pcmk__set_env_option(const char *option, const char *value)
     const char *const prefixes[] = {"PCMK_", "HA_"};
     char env_name[NAME_MAX];
 
+    CRM_CHECK(!pcmk__str_empty(option) && (strchr(option, '=') == NULL),
+              return);
+
     for (int i = 0; i < (sizeof(prefixes) / sizeof(char *)); i++) {
-        snprintf(env_name, NAME_MAX, "%s%s", prefixes[i], option);
-        if (value) {
+        int rv = snprintf(env_name, NAME_MAX, "%s%s", prefixes[i], option);
+
+        if (rv >= sizeof(env_name)) {
```

Comment:
I feel like `pcmk__set_env_option()` should return a boolean, indicating whether or not it successfully both variables. I didn't worry about it for this PR.

Reply:
bool should be reserved for true/false semantics; standard Pacemaker return codes should be used for success/failure, even if only two codes are possible.

Since it's an internal function, whether this one should have a return value should depend mainly on whether the callers should do anything differently based on success or failure.

Reply:
> return codes

Makes sense.

> do anything

I would expect at least some of them *should*, if we fail to set the env options that we're trying to set. Maybe, maybe not.

Reply:
It would be worthwhile to log errors inside the function itself, but since these tend to be for the benefit of resource agents, there's not much else to do